internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -126,7 +127,21 @@ func (ws *workflows) StartWorkflow(ctx context.Context, session sdktypes.Session
 
 func (ws *workflows) sessionWorkflow(wctx workflow.Context, params *sessionWorkflowParams) error {
 	wi := workflow.GetInfo(wctx)
-	sid := params.Data.Session.ID()
+	s := params.Data.Session
+	sid := s.ID()
+
+	// get session time (it was created not long after event dispatch) and then update it with actual event time if found
```

Comment:
you can't really assume that,
assume we get the event but workers are busy, you'll get a delay which might be long from event to event workflow
then you still need to call execute workflow for the session workflow, can have another delay
only then you get here

Reply:
assume what?

Reply:
can't assume the session time is not that different from event time

Reply:
we talked. I won't use session time as a sane default